pyodide-build/pyodide_build/recipe.py

```diff
@@ -0,0 +1,95 @@
+import functools
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Any
+
+from .io import MetaConfig
+
+
+@functools.lru_cache(maxsize=1)
+def load_all_recipes(recipe_dir: Path) -> dict[str, MetaConfig]:
+    """Load all package recipes from the recipe directory."""
+
+    recipes_path = recipe_dir.glob("*/meta.yaml")
+
+    recipes: dict[str, MetaConfig] = {}
+    for recipe in recipes_path:
+        try:
+            config = MetaConfig.from_yaml(recipe)
+            recipes[config.package.name] = config
+        except Exception as e:
+            raise ValueError(f"Could not parse {recipe}.") from e
+
+    return recipes
+
+
+def load_recipes(
+    recipe_dir: Path,
+    names_or_tags: Iterable[str],
+    load_always: bool = True,
+) -> tuple[dict[str, MetaConfig], dict[str, Any]]:
+    """
+    Load the recipes for the given package names or tags.
+    Note that this function does not do any dependency resolution.
+
+    Parameters
+    ----------
+    recipe_dir
+        Path to the recipe directory
+    names_or_tags
+        List of package names or tags to load.
+        It also supports the following special values:
+            - "*" : all packages
+            - "no-numpy-dependents" : all packages except those that depend on numpy (including numpy itself)
+    load_always
+        Whether to load packages with the "always" tag/
+
+    Returns
+    -------
+    recipes
+        Dictionary of package name => config
+    flags
```

Comment:
I don't fully understand why we need to additionally introduce the notion of flags. We could return the list of tags here, and check if `no-numpy-dependents` is in it, can't we? 

Or at least give it some more explicit name, because it's easy to confuse flags and tags, especially if one has a subset of keys of the other.

Reply:
I wanted to hide details about tags / meta-packages from buildall, but I think it was a bad choice. I removed flags and let `generate_dependency_graph` deal with no-numpy-dependents.